crates/gitbutler-hunk-dependency/src/diff.rs

```diff
@@ -0,0 +1,76 @@
+use anyhow::{anyhow, Context};
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct InputDiff {
+    pub old_start: u32,
+    pub old_lines: u32,
+    pub new_start: u32,
+    pub new_lines: u32,
+}
+
+impl InputDiff {
+    pub fn net_lines(&self) -> i32 {
+        self.new_lines as i32 - self.old_lines as i32
```

Comment:
You can use the do different kinds of subtraction as a method on the new_lines, and you wont have to cast.
You probably wanna do the 'checked' kind though, which returns an Option. 

Seems like net_lines is called from add_new which is called from add, which returns a Result. 
So you can do something like 
```rust
        self.net_lines()
            .checked_sub_unsigned(self.old_lines)
            .ok_or(anyhow!("Overflowed computing net lines"))
```
and do the ? operator. It's gonna be so much easier to debug in the future

Reply:
That function `checked_sub_unsigned` is only available on `i32`, but `new_lines` and `old_lines` are both `u32`. What can I do?

Reply:
I think what we'd want here is the nightly-only [`checked_signed_diff()`](https://doc.rust-lang.org/core/primitive.u32.html#method.checked_signed_diff), starting at `u32` and going to `i32` like what seems to be intended here.
Nightly might need to be updated in the toolchain file to make it available.

Reply:
ooh this is a really cool function